R/zchunk_L261.Cstorage.R

```diff
@@ -56,204 +68,319 @@ module_energy_L261.Cstorage_DISABLED <- function(command, ...) {
+      select(region, depresource = resource, subresource, grade, available, extractioncost) ->
+      L261.DepRsrcCurves_C # This is a final output table.
+
+    # Calculate three different supply curves of carbon storage resources: high, low, and lowest.
+    # Multiply the extraction cost by their respective cost multipliers.
+    HI_CCS_COST_MULT <- 0.8
+
+    LO_CCS_COST_MULT <- 3
+
+    LOWEST_CCS_COST_MULT <- 10
+
+    # Note that these will produce final output tables.
+    # High supply curves of carbon storage resources
+    L261.DepRsrcCurves_C_high <- mutate(L261.DepRsrcCurves_C, extractioncost = extractioncost * HI_CCS_COST_MULT)
+
+    # Low supply curves of carbon storage resources
+    L261.DepRsrcCurves_C_low <- mutate(L261.DepRsrcCurves_C, extractioncost = extractioncost * LO_CCS_COST_MULT)
+
+    # Lowest supply curves of carbon storage resources
+    L261.DepRsrcCurves_C_lowest <- mutate(L261.DepRsrcCurves_C, extractioncost = extractioncost * LOWEST_CCS_COST_MULT)
+
+
+    # C
+    # Carbon storage sector information
+    A61.sector %>%
+      write_to_all_regions(c(LEVEL2_DATA_NAMES[["Supplysector"]], "logit.type"),
+                           GCAM_region_names = GCAM_region_names) ->
+      L261.Supplysector_C # This is a final output table.
+
+
+    # D
+    # Subsector information
+
+    # Subsector logit exponents of carbon storage sector
+    A61.subsector_logit %>%
+      write_to_all_regions(c(LEVEL2_DATA_NAMES[["SubsectorLogit"]], "logit.type"),
+                           GCAM_region_names = GCAM_region_names) ->
+      L261.SubsectorLogit_C # This is a final output table.
+
+
+    # Subsector shareweights of carbon storage sectors
+    A61.subsector_shrwt %>%
+      write_to_all_regions(c(LEVEL2_DATA_NAMES[["SubsectorShrwtFllt"]], "logit.type"),
+                           GCAM_region_names = GCAM_region_names) ->
+      L261.SubsectorShrwtFllt_C # This is a final output table.
+
+
+    # E
+    # Technology information
+    # Identification of stub technologies of carbon storage
+    # A61.globaltech_shrwt reports carbon storage technology shareweights
+    # Note: assuming that the technology list in the shareweight table includes the full set (any others would default to a 0 shareweight)
+    A61.globaltech_shrwt %>%
+      write_to_all_regions(LEVEL2_DATA_NAMES[["Tech"]],
+                           GCAM_region_names = GCAM_region_names) %>%
+      select(region, supplysector, subsector, stub.technology = technology) ->
+      L261.StubTech_C # This is a final output table.
+
+    # Energy inputs and coefficients of global technologies for carbon storage
+    # A61.globaltech_coef reports carbon storage global technology coefficients
+    A61.globaltech_coef %>%
+      gather(year, value, matches(YEAR_PATTERN)) %>% # Convert to long form
+      mutate(year = as.integer(year), # Year needs to be integer or numeric to interpolate
+             value = as.numeric(value)) %>%
```

Comment:
I guess, if they're being reads as ints, why bother converting to numeric? This will happen implicitly if necessary in a calculation.